content/en/user-guide/tools/transparent-endpoint-injection/dns-server.md

```diff
@@ -8,52 +8,88 @@ aliases:
+
+### Fallback DNS server
+
+If you want to use another upstream DNS resolver than Google (default `8.8.8.8`),
+specify the fallback DNS server where all non-redirected queries (i.e., not matching `DNS_LOCAL_NAME_PATTERNS`) will be forwarded to:
 
 ```bash
-DNS_LOCAL_NAME_PATTERNS='.*(ecr|lambda).*.amazonaws.com'
+DNS_SERVER=1.1.1.1
 ```
 
-Using this configuration, the LocalStack DNS server only redirects ECR and Lambda domains to LocalStack, and the rest will be resolved via `$DNS_SERVER`. This can be used for hybrid setups, where certain API calls (e.g., ECR, Lambda) target LocalStack, whereas other services will target real AWS.
-
-{{< alert title="Warning" color="warning">}}
-We generally do not recommend connecting to real AWS from within LocalStack, in fact you should avoid using real AWS credentials anywhere in your LocalStack apps. Use this configuration with caution.
-{{< /alert >}}
+### Custom redirects
 
-There is the possibility to manually set the DNS server all not-redirected queries will be forwarded to:
+If you want to resolve only certain AWS URLs to LocalStack,
+specify a comma-separated list of hostname regex patterns such as:
 
 ```bash
-DNS_SERVER=1.1.1.1
+DNS_LOCAL_NAME_PATTERNS='.*(ecr|lambda).*.amazonaws.com'
 ```
 
-Per default, LocalStack uses the Google DNS resolver at `8.8.8.8`.
+Using this configuration, the LocalStack DNS server only resolves ECR and Lambda domains to LocalStack, and the rest will be resolved via the `DNS_SERVER` (i.e., the real DNS entry by default).
+This can be used for hybrid setups, where certain API calls (e.g., ECR, Lambda) target LocalStack, whereas other services will target real AWS.
+The regex pattern follows Python flavored-regex and can be tested at [regex101.com](https://regex101.com/r/OzIsQa/1).
+
+[The regex101 link is maintained by Joel Scheuner (requires linking to GitHub or Google account).
+It redirects to the main page if the saved example would not work.]: #
+
+{{< alert title="Warning" color="warning">}}
+Use this configuration with caution because we generally do not recommend connecting to real AWS from within LocalStack. 
+{{< /alert >}}
+
 
 ## Self-signed certificates
 
-When you configure transparent execution mode using DNS, you may still have to configure your application's AWS SDK to accept self-signed certificates.
-This is a technical limitation caused by the SSL certificate validation mechanism, due to the fact that we are repointing AWS domain names (e.g., `*.amazonaws.com`) to `localhost`. For example, the following command will fail with an SSL error:
+In the transparent execution mode using DNS in LocalStack Pro, you may still have to configure your application's AWS SDK to accept self-signed certificates.
+This is a technical limitation caused by the SSL certificate validation mechanism, due to the fact that we are repointing AWS domain names (e.g., `*.amazonaws.com`) to `localhost`.
```

Comment:
```suggestion
This is a technical limitation caused by the SSL certificate validation mechanism, due to the fact that we are repointing AWS domain names (e.g., `*.amazonaws.com`) to `localhost.localstack.cloud`.
```

Reply:
I don't like "technical limitation", it's a security feature ;)

Reply:
fixed `localhost` and re-phrased this part in the context of TEI